src/cmd/cmd.zig

```diff
@@ -435,13 +438,11 @@ fn validator() !void {
         logger.infof("incremental snapshot: {s}", .{inc_path});
     }
 
-    // cli parsing
-    const snapshot_dir_str = config.current.accounts_db.snapshot_dir;
-    const n_cpus = @as(u32, @truncate(try std.Thread.getCpuCount()));
-    var n_threads_snapshot_load: u32 = @intCast(config.current.accounts_db.num_threads_snapshot_load);
-    if (n_threads_snapshot_load == 0) {
-        n_threads_snapshot_load = n_cpus;
-    }
+    const n_threads_snapshot_load: u32 = blk: {
+        const n_threads_snapshot_load: u32 = config.current.accounts_db.num_threads_snapshot_load;
+        if (n_threads_snapshot_load != 0) break :blk n_threads_snapshot_load;
```

Comment:
nit: can we change this to be symmetric and have if statement check if == 0 -- its a lot easier to reason about things in the true case instead of the negative (ie, == 0 means only one value of x runs the branch, != 0 is inf values run the branch)